projects/ngrx.io/content/guide/effects/index.md

```diff
@@ -218,3 +218,103 @@ export class MovieModule {}
 **Note:** Running an effects class multiple times, either by `forRoot()` or `forFeature()`, (for example via different lazy loaded modules) will not cause Effects to run multiple times. There is no functional difference between effects loaded by `forRoot()` and `forFeature()`; the important difference between the functions is that `forRoot()` sets up the providers required for effects.
 
 </div>
+
+## Incorporating State
+
+If additional metadata is needed to perform an effect besides the initiating action's `type`, we should rely on passed metadata from an action creator's `props` method.
+
+Let's look at an example of an action initiating a login request using an effect with additional passed metadata:
+
+<code-example header="login-page.actions.ts">
+import { createAction, props } from '@ngrx/store';
+import { Credentials } from '@example-app/auth/models/user';
+
+export const login = createAction(
+  '[Login Page] Login',
+  props&lt;{ credentials: Credentials }&gt;()
+);
+
+export type LoginPageActionsUnion = ReturnType&lt;typeof login&gt;;
+</code-example>
+
+<code-example header="auth.effects.ts">
+import { Injectable } from '@angular/core';
+import { Actions, ofType, createEffect } from '@ngrx/effects';
+import { of } from 'rxjs';
+import { catchError, exhaustMap, map } from 'rxjs/operators';
+import {
+  LoginPageActions,
+  AuthApiActions,
+} from '@example-app/auth/actions';
+import { Credentials } from '@example-app/auth/models/user';
+import { AuthService } from '@example-app/auth/services/auth.service';
+
+@Injectable()
+export class AuthEffects {
+  login$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(LoginPageActions.login),
+      map(action => action.credentials),
```

Comment:
Personally I don't like to use `map` operator to extract the payload from the action and prefer to destruct the payload where needed.
How do we feel about this?

Reply:
I agree that's its not needed here and can be passed to the login method

Reply:
Agreed, I do the same.  Pulled this from the example-app, so wasn't sure if this was a "best practice".  Updated in 2a88984ac8f5874cf6910f953d1f42b9980d177b.